src/lsqecc/pauli_rotations/circuit.py

```diff
@@ -224,7 +224,7 @@ def commute_pi_over_four_rotation(self, index: int) -> None:
 
             # Flip the phase if product of coefficients is negative
             # Product of coefficients will always be either i or -i (see issues #28 for proof)
-            product_of_coefficients /= 1j
+            product_of_coefficients *= 1j
```

Comment:
I am 100% sure rn but I think the reason we use `*= 1j` is for checking whether we are getting i or -i.